test/peaks/yearn/YVaultController.test.js

```diff
@@ -0,0 +1,129 @@
+const assert = require('assert')
+const utils = require('../../utils.js')
+
+const toWei = web3.utils.toWei
+const fromWei = web3.utils.fromWei
+const toBN = web3.utils.BN
+const MAX = web3.utils.toTwosComplement(-1);
+const n_coins = 4
+
+/*
+Problems need to be solved
+
+1) How to populate controller with x amount of yCRV
+2) { from: this.yVaultPeak.address } => Error: sender not recognised
+    - Cannot have ganache account be a peak due to Address.isContract()
+    - How to test onlyPeak() if private key is not accessible
+3) assert.equal() for controller, peak and yVault should be simple after 1) is solved
+
+*/
+
+contract('YVaultController', async (accounts) => {
+
+    before(async () => {
+        const artifacts = await utils.getArtifacts()
+        Object.assign(this, artifacts)
+        this.owner = accounts[0]
+        /*
+        Init controller with 20 yCRV
+        */
+    })
+
+    // Owner tests
+    describe('Only Owner', async () => {
+        it('Owner can add Peak', async () => {
+            let reverted = false
+            try {
+                await this.controller.addPeak(this.yVaultPeak.address, {from: this.owner})
+            }
+            catch (e) {
+                reverted = true
```

Comment:
You'd need to assert the revert reason as well, otherwise, the tx could be reverting for another reason and this test will still pass. Same for other tests.